Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp

```diff
@@ -1010,5 +1041,56 @@ namespace EMotionFX
         {
             return HasEditorMaterialComponent() ? AZ::Edit::PropertyVisibility::Hide : AZ::Edit::PropertyVisibility::Show;
         }
+
+        void EditorActorComponent::DetachFromEntity()
+        {
+            if (!m_actorInstance)
+            {
+                return;
+            }
+
+            ActorInstance* attachedTo = m_actorInstance->GetAttachedTo();
+            if (attachedTo)
```

Comment:
nit/suggestion: We could move this inside the `if`

Reply:
Yeah I know about the c++17 if initializer thing. But somehow my brain still tell me to walk one stair at a time. Perhaps I will get use to that one day.

Reply:
I only like it as it reduces the scope so you can't accidentally misuse if `attachedTo`, `if` initializer can be nice for this as there are actually two statements instead of the implicit null check that happens without that